File: Counting x Hash Table/1726. Tuple with Same Product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) 
    {
        int ans = 0, n = nums.size();
        unordered_map<long, int> store;
        for(int i = 0; i < n; i++)
            for(int j = i + 1; j < n; j++)
                store[long(nums[i]) * nums[j]] += 1;

        for(int i = 0; i < n; i++)
            for(int j = i + 1; j < n; j++)
                ans += (store[long(nums[i]) * nums[j]] - 1) * 4;

        return ans; 
    }
};

File: Hash Table/1726. Tuple with Same Product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        ans, n = 0, len(nums)
        store = defaultdict(int)
        for i in range(n):
            for j in range(i + 1, n):
                store[nums[i] * nums[j]] += 1

        for i in range(n):
            for j in range(i + 1, n):
                ans += (store[nums[i] * nums[j]] - 1) * 4
        return ans